tests/hooks/test_subprocess.py

```diff
@@ -0,0 +1,110 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+from pathlib import Path
+from subprocess import PIPE, STDOUT
+from tempfile import TemporaryDirectory
+from unittest import mock
+
+import pytest
+from parameterized import parameterized
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.subprocess import SubprocessHook
+
+OS_ENV_KEY = 'SUBPROCESS_ENV_TEST'
+OS_ENV_VAL = 'this-is-from-os-environ'
+
+
+class TestSubprocessHook(unittest.TestCase):
+    @parameterized.expand(
+        [
+            ('with env', {'ABC': '123', 'AAA': '456'}, {'ABC': '123', 'AAA': '456', OS_ENV_KEY: ''}),
+            ('empty env', {}, {OS_ENV_KEY: ''}),
+            ('no env', None, {OS_ENV_KEY: OS_ENV_VAL}),
+        ]
+    )
+    def test_env(self, name, env, expected):
+        """
+        Test that env variables are exported correctly to the command environment.
+        When ``env`` is ``None``, ``os.environ`` should be passed to ``Popen``.
+        Otherwise, the variables in ``env`` should be available, and ``os.environ`` should not.
+        """
+        print(f"test_name: {name}")
```

Comment:
Intentional? Or just remnant from debugging ?

Reply:
Whoops miss that. Yes this was likely debugging that should be removed.

Reply:
resolved